### ISSUE TITLE
GL_INVALID_ENUM in glEnable(GL_DEPTH_CLAMP)

### ISSUE THREAD

**Issue by lurkermike** [gdm85/wolfengo#5]
Ubuntu 14.04 LTS
Go 1.6

I tried to get Wolfengo running in an Ubuntu VM after encountering the issues with OpenGL on OSX in issue #1 

I had to install the following packages: mesa-common-dev, libxrandr-dev, libxcursor-dev, libxinerama-dev, libgl1-mesa-dev, libxi-dev

Also, I had to make some minor tweaks to the Makefile to have an absolute GOBIN path rather than a relative path.  (Was getting the error "cannot install, GOBIN must be an absolute path" until I changed the GOBIN variable assignment.)

I am able to compile it successfully, but I get an error when running it:

```
WolfenGo v0.1.1, Copyright (C) 2016 gdm85
https://github.com/gdm85/wolfengo
WolfenGo comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under GNU/GPLv2 license.
2.1 Mesa 10.1.3
panic: [GL_DEBUG] source 33350 gltype 33356 id 1 severity 37190 length 43: GL_INVALID_ENUM in glEnable(GL_DEPTH_CLAMP)

goroutine 1 [running, locked to thread]:
panic(0x5fa420, 0xc82000a4e0)
    /usr/src/go/src/runtime/panic.go:464 +0x3e6
main.debugCb(0x824c00008246, 0x914600000001, 0x2b, 0xc82000e360, 0x2b, 0x0)
    /home/mike/go/src/wolfengo/src/main.go:69 +0x39f
wolfengo/vendor/github.com/go-gl/gl/v2.1/gl.glowDebugCallback_gl21(0x824c00008246, 0x914600000001, 0xc80000002b, 0x7fff9fca34a0, 0x0)
    /home/mike/go/src/wolfengo/vendor/github.com/go-gl/gl/v2.1/gl/debug.go:29 +0x8e
wolfengo/vendor/github.com/go-gl/gl/v2.1/gl._cgoexpwrap_4b34acd434eb_glowDebugCallback_gl21(0x824c00008246, 0x914600000001, 0x2b, 0x7fff9fca34a0, 0x0)
    wolfengo/vendor/github.com/go-gl/gl/v2.1/gl/_obj/_cgo_gotypes.go:45785 +0x53
wolfengo/vendor/github.com/go-gl/gl/v2.1/gl._Cfunc_glowEnable(0x7f9eff47fb60, 0x3030000864f)
    wolfengo/vendor/github.com/go-gl/gl/v2.1/gl/_obj/_cgo_gotypes.go:12707 +0x3a
wolfengo/vendor/github.com/go-gl/gl/v2.1/gl.Enable(0x3030000864f)
    /home/mike/go/src/wolfengo/vendor/github.com/go-gl/gl/v2.1/gl/package.go:19502 +0x2b
main.main()
    /home/mike/go/src/wolfengo/src/main.go:124 +0x45c
```

**Comment by lurkermike** [comment]
After I opened this, I poked around for closed issues and found issue #2 about debug being enabled causing the crashes. I set debugGL to false and recompiled.  Got another error this time:

```
panic: runtime error: cgo argument has Go pointer to Go pointer

goroutine 1 [running, locked to thread]:
panic(0x68b260, 0xc8203c0500)
    /usr/src/go/src/runtime/panic.go:464 +0x3e6
wolfengo/vendor/github.com/go-gl/gl/v2.1/gl.ShaderSource(0x100000002, 0xc820024098, 0x0)
    /home/mike/go/src/wolfengo/vendor/github.com/go-gl/gl/v2.1/gl/package.go:24310 +0x4c
main.(*Shader).addProgram(0xc8203c04f0, 0xc820054dd0, 0xcf, 0x8b31, 0x0, 0x0)
    /home/mike/go/src/wolfengo/src/shader.go:104 +0x1c2
main.(*Shader).addProgramFromFile(0xc8203c04f0, 0x6f9af0, 0x11, 0x8b31, 0x0, 0x0)
    /home/mike/go/src/wolfengo/src/shader.go:123 +0x123
main.getBasicShader(0x30300000de1, 0x0, 0x0)
    /home/mike/go/src/wolfengo/src/level.go:73 +0x130
main.main()
    /home/mike/go/src/wolfengo/src/main.go:130 +0x448
```

Which, from reading issue #1 this is related to go-gl not supporting Go 1.6 yet.

So, sorry for the false alarm!  Hopefully this helps someone else.

**Comment by gdm85** [comment]
No worries, I am afraid Go and OpenGL still need some time to get along together smoothly..
